kibana-reports/public/components/report_definitions/report_settings/report_settings.tsx

```diff
@@ -442,67 +448,169 @@ export function ReportSettings(props) {
     return dashboard_options;
   };
 
-  const getDashboardBaseUrl = () => {
+  const getDashboardBaseUrlCreate = () => {
     let baseUrl = window.location.href;
+    if (edit) {
+      return baseUrl.replace(
+        `opendistro_kibana_reports#/edit/${editDefinitionId}`,
+        'dashboards#/view/'
+      );
+    }
     return baseUrl.replace(
       'opendistro_kibana_reports#/create',
       'kibana#/dashboard/'
     );
   };
 
-  const getVisualizationOptions = (data) => {
-    let index;
-    let options = [];
-    for (index = 0; index < data.length; ++index) {
-      let entry = {
-        value: data[index]['_id'].substring(14),
-        text: data[index]['_source']['visualization']['title'],
-      };
-      options.push(entry);
+  const setReportSourceDropdownOption = (options, reportSource, url) => {
+    let index = 0;
+    if (reportSource === reportSourceType.dashboard) {
+      for (index = 0; index < options.dashboard.length; ++index) {
+        if (url.includes(options.dashboard[index].value)) {
+          setDashboardSourceSelect(options.dashboard[index].value);
+        }
+      }
+    } else if (reportSource === reportSourceType.visualization) {
+      for (index = 0; index < options.visualizations.length; ++index) {
+        if (url.includes(options.visualizations[index].value)) {
+          setVisualizationSourceSelect(options.visualizations[index].value);
+        }
+      }
+    } else if (reportSource === reportSourceType.savedSearch) {
+      // todo: add logic once get Visualizations + search is merged
```

Comment:
can we get rid of this since that PR was merged?

Reply:
Yeah forgot to add saved search logic, will add in